thunder/tests/opinfos.py

```diff
@@ -2493,6 +2493,54 @@ def where_sample_generator(op, device, dtype, requires_grad, **kwargs):
 conditional_and_mask_ops.append(where_opinfo)
 
 
+def nan_to_num_sample_generator(op, device, dtype, requires_grad, **kwargs):
+    make = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    cases = (
+        ((2, 1, 2), None, None, None),
+        ((4, 4), None, 1.0, None),
+        ((4, 4), None, None, 1.0),
+        ((5,), None, 1.0, 0.0),
+        ((8, 1, 6), 1, None, None),
+        ((8, 7, 5, 1), 1, 1.0, None),
+        ((8, 7, 5, 1), 1, None, 0.0),
+        ((8, 7, 5, 1), 1, 1.0, 0.0),
+    )
+
+    for a_shape, nan, posinf, neginf in cases:
+        yield SampleInput(make(a_shape, dtype=dtype, requires_grad=requires_grad), nan, posinf, neginf)
+
+
+def nan_to_num_error_generator(op, device, dtype=torch.float32, **kwargs):
+    make = partial(make_tensor, device=device, dtype=dtype)
+    err_msg = "out is not None which is currently unsupported"
+    yield (
+        SampleInput(
+            make(
+                5,
+            ),
+            None,
+            None,
+            None,
+            make(
+                5,
+            ),
+        ),
+        NotImplementedError,
+        err_msg,
+    )
+
+
+nan_to_num_opinfo = OpInfo(
+    ltorch.nan_to_num,
+    supports_grad=True,
```

Comment:
The test failures are due to missing functionality in the "phantom grad" transform, and those tests are controlled by this setting (which has a very confusing name!). fyi @IvanYashchuk, who's looking at unifying grad transforms, to maybe remove/replace/rename this setting so it's not so confusing.

If this line is removed and the test inputs are updated to include extremal values, then I think this PR will work as expected. Let's make those updates and mark it ready for review!